### PR TITLE
[routing-manager] `GetFavoredOmrPrefix` should return an error when Routing Manager is not running

### PR DESCRIPTION
**Pull request by superwhd**

### CODE REVIEW

File: src/core/border_router/routing_manager.cpp [hunk 1]
```diff
@@ -149,6 +149,7 @@ Error RoutingManager::GetFavoredOmrPrefix(Ip6::Prefix &aPrefix, RoutePreference
     Error error = kErrorNone;
 
     VerifyOrExit(IsInitialized(), error = kErrorInvalidState);
+    VerifyOrExit(Get<Mle::Mle>().IsAttached(), error = kErrorInvalidState);
```
Comment (wgtdkp) [inline src/core/border_router/routing_manager.cpp:152]: We should check against the RoutingManager status. For example, this is only available when `mIsRunning` is true?

Comment (superwhd) [reply src/core/border_router/routing_manager.cpp]: Good point!

File: include/openthread/border_routing.h [hunk 2]
```diff
@@ -190,7 +190,7 @@ otError otBorderRoutingGetOmrPrefix(otInstance *aInstance, otIp6Prefix *aPrefix)
  * @param[out]  aPrefix      A pointer to output the favored OMR prefix.
  * @param[out]  aPreference  A pointer to output the preference associated the favored prefix.
  *
- * @retval  OT_ERROR_INVALID_STATE  The Border Routing Manager is not initialized yet.
+ * @retval  OT_ERROR_INVALID_STATE  The Border Routing Manager is not running yet.
```
Comment (abtink) [inline include/openthread/border_routing.h:193]: Update the doc in `routing_manager.hpp` as well?

**Review by abtink — APPROVED** [review]
LGTM 👍

File: include/openthread/instance.h [hunk 3]
```diff
@@ -53,7 +53,7 @@ extern "C" {
  * @note This number versions both OpenThread platform and user APIs.
  *
  */
-#define OPENTHREAD_API_VERSION (270)
+#define OPENTHREAD_API_VERSION (271)
```
Comment (jwhui) [inline include/openthread/instance.h:56]: Please rebase and increment again.

**Review by jwhui — APPROVED** [review]

**Pull request merged**